os/hal/ports/AT32/AT32F41x/hal_lld.c

```diff
@@ -0,0 +1,261 @@
+
+  /* Initializes the backup domain. */
+  hal_lld_battery_powered_domain_init();
+
+  /* DMA subsystems initialization. */
+#if defined(AT32_DMA_REQUIRED)
+  dmaInit();
+#endif
+
+  /* IRQ subsystem initialization. */
+  irqInit();
+
+  /* Power voltage monitoring enable. */
+#if AT32_PVM_ENABLE
+  PWC->CTRL |= PWC_CTRL_PVMEN | (AT32_PVMSEL & AT32_PVMSEL_MASK);
+#endif /* AT32_PVM_ENABLE */
+}
+
+/**
+ * @brief   AT32 clocks and PLL initialization.
+ * @note    All the involved constants come from the file @p board.h.
+ * @note    This function should be invoked just after the system reset.
+ *
+ * @special
+ */
+
+/*
+ * Clocks initialization for all series.
+ */
+void at32_clock_init(void) {
+
+#if !AT32_NO_INIT
+  /* HICK setup, it enforces the reset situation in order to handle possible
+     problems with JTAG probes and re-initializations. */
+  CRM->CTRL |= CRM_CTRL_HICKEN;             /* Make sure HICK is ON.          */
+  while (!(CRM->CTRL & CRM_CTRL_HICKSTBL))
+    ;                                       /* Wait until HICK is stable.     */
+
+  /* HICK is selected as new source without touching the other fields in
+     CFGR. Clearing the register has to be postponed after HICK is the
+     new source. */
+  CRM->CFG &= ~CRM_CFG_SCLKSEL;             /* Reset SCLKSEL, selecting HICK. */
+  while ((CRM->CFG & CRM_CFG_SCLKSTS) != CRM_CFG_SCLKSTS_HICK)
+    ;                                       /* Waits until HICK is selected.  */
+
+  /* Registers finally cleared to reset values. */
+  CRM->CTRL &= ~(0x010D0000U);              /* CTRL reset value.             */
+  CRM->CFG = 0x00000000;                    /* CFG reset value.              */
+  CRM->PLL = 0x00001F10U;                   /* PLL reset value.              */
+  CRM->MISC1 = 0x00100000;                  /* MISC1 reset value.            */
+  CRM->MISC2 = 0x0000000D;                  /* MISC2 reset value.            */
+  CRM->CLKINT = 0x009F0000;                 /* CLKINT reset value.           */
+
+#if AT32_HEXT_ENABLED
+#if defined(AT32_HEXT_BYPASS)
+  /* HEXT Bypass. */
+  CRM->CTRL |= CRM_CTRL_HEXTEN | CRM_CTRL_HEXTBYPS;
+#endif
+  /* HEXT activation. */
+  CRM->CTRL |= CRM_CTRL_HEXTEN;
+  while (!(CRM->CTRL & CRM_CTRL_HEXTSTBL))
+    ;                                       /* Waits until HEXT is stable.    */
+#endif
+
+#if AT32_LICK_ENABLED
+  /* LICK activation. */
+  CRM->CTRLSTS |= CRM_CTRLSTS_LICKEN;
+  while ((CRM->CTRLSTS & CRM_CTRLSTS_LICKSTBL) == 0)
+    ;                                       /* Waits until LICK is stable.    */
+#endif
+
+#if AT32_ACTIVATE_PLL
+  /* PLL activation. */
+#if (AT32_PLLCFGEN == AT32_PLLCFGEN_SOLID)
+  /* Solid PLL config. */
+  CRM->CFG |= AT32_PLLMULT | AT32_PLLHEXTDIV | AT32_PLLRCS;
+#ifdef AT32_PLLCLKREF
+  CRM->PLL |= AT32_PLLCLKREF;
+#endif
+#else
+  /* Flexible PLL config. */
+  CRM->CFG  |= AT32_PLLHEXTDIV | AT32_PLLRCS;
+  CRM->PLL  |= AT32_PLL_FR | AT32_PLL_MS | AT32_PLL_NS | AT32_PLLCFGEN;
```

Comment:
Change it to:
```
CRM->PLL  = AT32_PLL_FR | AT32_PLL_MS | AT32_PLL_NS | AT32_PLLCFGEN;
```
or add forward:
```
CRM->PLL &= ~(AT32_PLL_MS_MASK | AT32_PLL_MS_MASK );
```
Because the reset value of REG PLL is 0x00001F10.

Reply:
Yeah, I've just try this and it work perfectly.
```
CRM->PLL  = AT32_PLL_FR | AT32_PLL_MS | AT32_PLL_NS | AT32_PLLCFGEN;
```
Testing with these setup:
- 8 MHz HEXT -> PLL flexible -> PLL_MS_DIV_1 -> PLL_NS_MULT_96 -> PLL_FR_DIV_16
- 8 MHz HEXT -> PLL flexible -> PLL_MS_DIV_2 -> PLL_NS_MULT_192 -> PLL_FR_DIV_16
- 12 MHz HEXT -> PLL flexible -> PLL_MS_DIV_1 -> PLL_NS_MULT_64 -> PLL_FR_DIV_16
- 12 MHz HEXT -> PLL flexible -> PLL_MS_DIV_2 -> PLL_NS_MULT_128 -> PLL_FR_DIV_16
- 4 MHz HICK -> PLL flexible -> PLL_MS_DIV_1 -> PLL_NS_MULT_240 -> PLL_FR_DIV_8
- 4 MHz HICK -> PLL flexible -> PLL_MS_DIV_2 -> PLL_NS_MULT_384 -> PLL_FR_DIV_16